components/dot_menu/dot_menu.tsx

```diff
@@ -112,19 +130,22 @@ export default class DotMenu extends React.PureComponent {
             openUp: false,
             width: 0,
             canEdit: props.canEdit && !props.isReadOnly,
+            canDelete: props.canDelete && !props.isReadOnly,
         };
 
         this.buttonRef = React.createRef();
     }
 
     disableCanEditPostByTime() {
-        const {post, isLicensed, postEditTimeLimit} = this.props;
+        const {post, isLicensed} = this.props;
         const {canEdit} = this.state;
 
+        const postEditTimeLimit = this.props.postEditTimeLimit || String(Constants.UNSET_POST_EDIT_TIME_LIMIT);
```

Comment:
Could you parse the time limit in the `mapStateToProps` and just pass in a number? That's kind of gross that the old code worked here